stateoftheuniverse/widgets/moon_functions.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/python3
+"""
+A function to return moon phase & moon brightness.
+"""
+
+import numpy as np
+import astroplan
+import astropy
+from astropy.time import Time
+import matplotlib.pyplot as plt
+from astropy.coordinates import get_moon, get_sun
+import datetime
+import warnings
+warnings.filterwarnings('ignore')
+
+def get_moon_illumination(time: datetime) -> float:
+    """ A function to return the moon illumination
+
+    I'll fill this in later
+    """
+
+    astropytime = Time(time)
+    try:
+        illumination = astroplan.moon.moon_illumination(astropytime)
+    except:
+        print("Something has gone wrong. Please raise an issue on GitHub")
+    return illumination*100
+
+def get_moon_state(time: datetime) -> bool:
+    """ A function to return whether the moon is waxing or waning
+    """
+    astropytime = Time(time)
+    moonra = get_moon(time).ra.value
+    sunra = get_sun(time).ra.value
+    opposite = sunra + 180.
+    if opposite > 360.:
+        opposite -= 360.
+
+    #Moon goes for west to east
+    #Find whether waxing or waning
+    if sunra > 180:
+        if (moonra > sunra) or (moonra < opposite):
+            waxing = True
+        else:
+            waxing=False
+
+    elif sunra <= 180:
+        if (moonra > sunra) & (moonra < opposite):
+            waxing = True
+        else:
+            waxing=False
+
+def get_moon_phase(time: datetime) -> str:
+    """ Code to get moon phase"""
+    illumination = get_moon_illumination(time)
+    waxing = get_moon_state(time)
+
+    if  illumination <= 0.1:
+        phase = 'New Moon'
+    if (illumination > 0.1) & (illumination < 49.9):
+        if waxing:
+            phase = 'Waxing Crescent'
+        if not waxing:
+            phase = 'Waning Crescent'
+    if (illumination >= 49.9) & (illumination <= 50.1):
+        if waxing:
+            phase = 'First Quarter'
+        if not waxing:
+            phase = 'Last Quarter'
+    if (illumination > 50.1) & (illumination < 99.9):
+        if waxing:
+            phase = 'Waxing Gibbous'
+        if not waxing:
+            phase = 'Waning Gibbous'
+    if  illumination >= 99.9:
+        phase = 'Full Moon'
+
+    return phase
+
+if __name__ == "__main__":
+    time = Time(datetime.datetime.now())
+    print('{:.2f} %'.format(get_moon_illumination(time)))
```

Comment:
Just in case you don't know about f-strings already, check out the [PEP 498](https://www.python.org/dev/peps/pep-0498/) on literal string interpolation. Starting with Python 3.6, this feature allows you to write this line as:
```
print(f'{get_moon_illumination(time):.2f} %')
```
which is maybe a bit easier to read?

Reply:
Ooh I've been meaning to learn fstrings but never got round to it. Thanks for the example! Will implement later!